annotations/static/annotations/js/annotators/appellationlist.js

```diff
@@ -1,101 +1,283 @@
+						<span v-if="appellation.visible" class="glyphicon glyphicon glyphicon-eye-open"></span>
+						<span v-else class="glyphicon glyphicon glyphicon-eye-close"></span>
+					</a>
+				</span>
+				
+				{{ label() }}
+				<div class="text-warning">
+					<input v-if="sidebar == 'submitAllAppellations'" type="checkbox" v-model="checked" aria-label="...">
+					Created by <strong>{{ getCreatorName(appellation.createdBy) }}</strong> on {{ getFormattedDate(appellation.created) }}
+				</div>
+				</li>`,
+	data: function () {
+		return {
+			checked: true,
+			canUncheckAll: false,
+			canCheckAll: false
+		}
+	},
+	mounted: function () {
+		this.watchUncheckStore();
+		this.watchCheckStore();
+	},
+	watch: {
+		checked: function () {
+			if (this.checked == false) {
+				store.commit('removeAppellation', this.index);
+				store.commit('setSelectFalse')
+			} else {
+				store.commit('addAppellation', this.appellation)
+				store.commit('setDeselectFalse')
+			}
+		},
+
+	},
+	methods: {
+		watchUncheckStore: function () {
+			store.watch(
+				(state) => {
+					return store.getters.getDeselect
+				},
+				(val) => {
+					if (val) {
+						this.uncheckAll()
+						this.canCheckAll = true;
+					} else {
+						console.log('else');
+					}
+				},
+			);
+		},
+		watchCheckStore: function () {
+			store.watch(
+				(state) => {
+					return store.getters.getSelect
+				},
+				(val) => {
+					if (val) {
+						this.checkAll()
+					}
+				},
+			);
+		},
+
+		uncheckAll: function () {
+			this.checked = false;
+		},
+		checkAll: function () {
+			this.checked = true;
+		},
+		hide: function () {
+			this.$emit("hideappellation", this.appellation);
+		},
+		show: function () {
+			this.$emit("showappellation", this.appellation);
+		},
+		toggle: function () {
+			if (this.appellation.visible) {
+				this.hide();
+			} else {
+				this.show();
+			}
+		},
+		isSelected: function () {
+			return this.appellation.selected;
+		},
+		select: function () {
+			this.$emit('selectappellation', this.appellation);
+		},
+		label: function () {
+			if (this.appellation.interpretation) {
+				return this.appellation.interpretation.label;
+			} else if (this.appellation.dateRepresentation) {
+				return this.appellation.dateRepresentation;
+			}
+		},
+		getCreatorName: function (creator) {
+			if (creator.id == USER_ID) {
+				return 'you';
+			} else {
+				return creator.username;
+			}
+		},
+		getFormattedDate: function (isodate) {
+			var date = new Date(isodate);
+			var monthNames = [
+				"January", "February", "March",
+				"April", "May", "June", "July",
+				"August", "September", "October",
+				"November", "December"
+			];
+			var minutes = String(date.getMinutes());
+			if (minutes.length == 1) {
+				minutes = '0' + minutes;
+			}
+
+			var day = date.getDate();
+			var monthIndex = date.getMonth();
+			var year = date.getFullYear();
+
+			return day + ' ' + monthNames[monthIndex] + ', ' + year + ' at ' + date.getHours() + ':' + minutes;
```

Comment:
I know this code is not new and used in other places, but does there seem to be any reason not to use a js library like moment.js? https://momentjs.com/